### DISCUSSION TITLE
Deploy a Fullstack App

### DISCUSSION THREAD

**Discussion by h-li** [prisma/prisma#21526] [Q&A] [answered]
### Question

I followed this guide [How to Build a Fullstack App with Next.js, Prisma, and Vercel Postgres](https://vercel.com/guides/nextjs-prisma-postgres) and am deploying the app on Vercel. Then I got this error:

Failed to compile.
--
13:51:10.016 |  
13:51:10.016 | ./node_modules/@prisma/client/runtime/library.d.ts:1286:35
13:51:10.016 | Type error: '?' expected.
13:51:10.016 |  
13:51:10.016 | 1284 \|
13:51:10.016 | 1285 \| export declare type GetFindResult<P extends OperationPayload, A> = Equals<A, any> extends 1 ? DefaultSelection<P> : A extends {
13:51:10.016 | > 1286 \|     select: infer S extends object;
13:51:10.016 | \|                                   ^
13:51:10.017 | 1287 \| } & Record<string, unknown> \| {
13:51:10.017 | 1288 \|     include: infer I extends object;
13:51:10.017 | 1289 \| } & Record<string, unknown> ? {
13:51:10.043 |  
13:51:10.043 | > Build error occurred
13:51:10.046 | Error: Call retries were exceeded
13:51:10.046 | at ChildProcessWorker.initialize (/vercel/path0/node_modules/next/dist/compiled/jest-worker/index.js:1:11661)
13:51:10.046 | at ChildProcessWorker._onExit (/vercel/path0/node_modules/next/dist/compiled/jest-worker/index.js:1:12599)
13:51:10.046 | at ChildProcess.emit (node:events:514:28)
13:51:10.046 | at ChildProcess._handle.onexit (node:internal/child_process:291:12) {
13:51:10.047 | type: 'WorkerError'
13:51:10.047 | }
13:51:10.059 | Error: Command "npx prisma generate && next build" exited with 1

What problem can this possibly indicate? It seems to be talking about Prisma Client runtime library. How can I fix it? I wasn't able to find any discussion that had had this issue.

### How to reproduce (optional)

Follow the guide [How to Build a Fullstack App with Next.js, Prisma, and Vercel Postgres](https://vercel.com/guides/nextjs-prisma-postgres) or here is [the app in my repo](https://github.com/h-li/prisma). Try deploying it on Vercel and you'll be able to reproduce.

### Expected behavior (optional)

Success deployment on Vercel with database connected.

### Information about Prisma Schema, Client Queries and Environment (optional)

All this information was followed exactly identically to how the guide said. 
When deploying, I've added the env vars they said needed in the guide.

**Comment by h-li** [comment, marked as answer]
Sorry for bothering you everyone = = turned out that

You need TypeScript 5.0+ to be compatible with Prisma Client 5.

Happy coding!

**Reply by janpio** [reply]
That should not be necessary actually, even the newest versions should still work with 4.7+:
https://www.prisma.io/docs/reference/system-requirements#software-requirements

But great you found a way to make it work.
